x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Types.java

```diff
@@ -113,4 +114,21 @@ private static boolean boolSetting(Object value, boolean defaultValue) {
     private static int intSetting(Object value, int defaultValue) {
         return value == null ? defaultValue : Integer.parseInt(value.toString());
     }
+    
+    private static void propagateUnsupportedType(String inherited, String originalType, Map<String, EsField> properties) {
+        if (properties != null && properties.isEmpty() == false) {
+            for (Entry<String, EsField> entry : properties.entrySet()) {
+                EsField field = entry.getValue();
+                UnsupportedEsField u;
+                if (field instanceof UnsupportedEsField) {
```

Comment:
Why is the `if` needed - the result will be the same regardless of whether field is `UnsupportedEsField` or not.